frontend/src/components/tabs/MetricsTab.tsx

```diff
@@ -0,0 +1,194 @@
+      <div className={padding(20)}>
+        <div>
+          This step corresponds to execution{' '}
+          <Link
+            className={commonCss.link}
+            to={RoutePageFactory.executionDetails(execution.getId())}
+          >
+            "{ExecutionHelpers.getName(execution)}".
+          </Link>
+        </div>
+        {isLoading && <Banner message='Metrics is loading.' mode='info' />}
+        {errorArtifacts && (
+          <Banner
+            message='Error in retriving metrics information.'
+            mode='error'
+            additionalInfo={errorArtifacts.message}
+          />
+        )}
+        {errorArtifactTypes && (
+          <Banner
+            message='Error in retriving artifact types information.'
+            mode='error'
+            additionalInfo={errorArtifactTypes.message}
+          />
+        )}
+        {!isLoading && !errorArtifacts && !errorArtifactTypes && artifacts && artifactTypes && (
+          <MetricsSwitcher artifacts={artifacts} artifactTypes={artifactTypes}></MetricsSwitcher>
+        )}
+      </div>
+    </div>
+  );
+}
+
+/**
+ * Visualize system metrics based on artifact input.
+ */
+function MetricsSwitcher({ artifacts, artifactTypes }: MetricsSwitcherProps) {
+  // system.ClassificationMetrics contains confusionMatrix or confidenceMetrics.
+  // TODO: Visualize confusionMatrix using system.ClassificationMetrics artifacts.
+  // https://github.com/kubeflow/pipelines/issues/5668
+  let classificationMetricsArtifacts = filterArtifactsByType(
+    'system.ClassificationMetrics',
+    artifactTypes,
+    artifacts,
+  );
+
+  const confidenceMetricsArtifacts = classificationMetricsArtifacts
+    .map(artifact => ({
+      name: artifact
+        .getCustomPropertiesMap()
+        .get('name')
+        ?.getStringValue(),
+      customProperties: artifact.getCustomPropertiesMap(),
+    }))
+    .filter(x => !!x.name)
+    .filter(x => {
+      const confidenceMetrics = x.customProperties
+        .get('confidenceMetrics')
+        ?.getStructValue()
+        ?.toJavaScript();
+      return !!confidenceMetrics;
+    });
+
+  return (
+    <>
+      {(!confidenceMetricsArtifacts || confidenceMetricsArtifacts.length === 0) && (
+        <Banner message='There is no metrics artifact available in this step.' mode='info' />
+      )}
+      {classificationMetricsArtifacts &&
+        classificationMetricsArtifacts.length > 0 &&
+        classificationMetricsArtifacts.map(artifact => {
+          const id = artifact.getId();
+          const customProperties = artifact.getCustomPropertiesMap();
+          const confidenceMetrics = customProperties
+            .get('confidenceMetrics')
+            ?.getStructValue()
+            ?.toJavaScript();
+          return (
+            <React.Fragment key={id}>
+              {confidenceMetrics && (
+                <div>
+                  <div className={padding(40, 'lr')}>
+                    <h1>
+                      ROC Curve{' '}
```

Comment:
shall we also show artifact name in the title?
you can get name by `customProperties.get('name').getStringValue()`

Reply:
I have thought about it, is the name of this always `metrics`? If so, I am not sure how much information we are adding here. For now let me add this name first.